content/terraform/v1.12.x/docs/language/parameterize/configure.mdx

```diff
@@ -0,0 +1,428 @@
+- Pass information from a Terraform operation to an automation tool.
+
+Add `output` blocks to export information about your module. For example, you can add two `output` blocks to expose the ID and the IP address of your configuration's web server:
+
+<CodeBlockConfig highlight="1,6" hideClipboard>
+
+```hcl
+output "instance_id" {
+  description = "ID of the EC2 instance"
+  value       = aws_instance.web.id
+}
+
+output "instance_ip" {
+  description = "Private IP address of the EC2 instance"
+  value       = aws_instance.web.private_ip
+}
+
+resource "aws_instance" "web" {
+  # …
+}
+```
+
+</CodeBlockConfig>
+
+Depending on whether an output value is defined in your root module or a child module, you can access the information exposed in different ways. Terraform displays root module output values in the CLI after you apply your configuration, and if you are using HCP Terraform, your workspace's overview page lists your configuration's outputs.
+
+You can use outputs to pass values from a child module to a parent module, and parent modules access those outputs using `module.<CHILD_MODULE_NAME>.<OUTPUT_NAME>` syntax.
+
+For example, your parent module can access the `instance_ip` and `instance_id` outputs from a child module named `web_server` by referencing `module.web_server.NAME`:
+
+<CodeBlockConfig highlight="10,20" hideClipboard>
+
+```hcl
+module "web_server" {
+  source = "./modules/web_server"
+  # …
+}
+
+resource "aws_route53_record" "web" {
+  zone_id = data.aws_route53_zone.main.zone_id
+  name    = "web.example.com"
+  type    = "A"
+  records = [module.web_server.instance_ip]
+  #...
+}
+
+resource "aws_cloudwatch_alarm" "web_health" {
+  alarm_name          = "web-server-health"
+  comparison_operator = "GreaterThanThreshold"
+  threshold           = "80"
+
+  dimensions = {
+    InstanceId = module.web_server.instance_id
+  }
+  # ...
+}
+```
+
+</CodeBlockConfig>
+
+If you are outputting sensitive data such as a password or API key, use the `sensitive` argument to prevent Terraform from displaying the value in CLI output:
+
+<CodeBlockConfig highlight="4" hideClipboard>
+
+```hcl
+output "database_password" {
+  description = "Auto-generated password for the RDS database instance"
+  value       = aws_db_instance.main.password
+  sensitive   = true
+}
+```
+
+</CodeBlockConfig>
+
+Terraform does still store the values of sensitive outputs in your state, and if you use the [`terraform output` CLI command](/terraform/cli/commands/output) with the `-json` or `-raw` flags then Terraform displays sensitive outputs in plain text.
+
+Adding the `ephemeral` argument to an output omits that value from state and plan files all together, but adds restrictions to the values you can assign that output. To learn more about the `ephemeral` argument and handling and outputting sensitive data in your configuration, refer to [Sensitive data in state](/terraform/language/state/sensitive-data).
```

Comment:
```suggestion
Adding the `ephemeral` argument to an output omits that value from state and plan files, but it adds restrictions to the values you can assign that output. To learn more about the `ephemeral` argument and handling and outputting sensitive data in your configuration, refer to [Sensitive data in state](/terraform/language/state/sensitive-data).
```
This might be a better suggestion than the one in line 115.